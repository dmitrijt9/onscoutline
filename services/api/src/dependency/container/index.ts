import { Connection, getCustomRepository } from 'typeorm'
import { CompetitionRepository } from '../../repositories/CompetitionRepository'
import { FacrScraper } from '../../services/scrapers/FacrScraper'
import { Scraper } from '../../services/scrapers/Scraper'
import { AppConfig, getAppConfig } from '../config/index'
import { bootstrapDbConnection } from './bootstrap/db-connection'

export const createContainer = async (
    appConfig = getAppConfig(process.env),
): Promise<Container> => {
    const typeormConnection = await bootstrapDbConnection(appConfig)

    const competitionRepository = getCustomRepository(CompetitionRepository)

    const scraper = new Scraper()
    const facrScraper = new FacrScraper(scraper, appConfig, competitionRepository)

    return {
        config: appConfig,
        typeormConnection,
        competitionRepository,
        facrScraper: facrScraper,
    }
}

export interface Container {
    config: AppConfig
    typeormConnection: Connection
    competitionRepository: CompetitionRepository
    facrScraper: FacrScraper
}
