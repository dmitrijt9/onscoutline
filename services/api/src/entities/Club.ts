import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
export class Club {
    @PrimaryGeneratedColumn()
    id: number

    @Column('varchar', { unique: true, nullable: true })
    facrId?: string

    @Column()
    name: string

    @Column('varchar', { nullable: true })
    facrUuid?: string
}
