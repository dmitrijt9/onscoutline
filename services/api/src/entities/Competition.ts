import { Column, Entity, Index, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
@Index('competition_facr_id', ['regionId', 'facrId'])
export class Competition {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    regionId: string

    @Column()
    regionName: string

    @Column()
    facrId: string

    @Column()
    facrUuid: string
}
