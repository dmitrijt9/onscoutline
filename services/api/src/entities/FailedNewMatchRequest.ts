import { NewMatchRequest } from '../services/match/types'
import { jsonTransformer } from '../utils/typeorm/jsonTransformer'
import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm'

@Entity({})
export class FailedNewMatchRequest {
    @PrimaryGeneratedColumn('uuid')
    id: string

    @Column()
    matchFacrUuid: string

    @Column('varchar')
    status: FailedNewMatchRequestStatus

    @Column('longtext', { transformer: jsonTransformer('requestJson') })
    requestJson: NewMatchRequest

    @CreateDateColumn()
    createdAt?: string

    @UpdateDateColumn()
    updatedAt?: string
}

export enum FailedNewMatchRequestStatus {
    Unprocessed = 'Unprocessed',
    UnprocessedClub = 'UnprocessedClub',
    UnprocessedPlayer = 'UnprocessedPlayer',
    Processed = 'Processed',
}
