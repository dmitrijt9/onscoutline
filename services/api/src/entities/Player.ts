import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
export class Player {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    surname: string

    @Column('date')
    dateOfBirth: string

    @Column('date')
    facrMemberFrom: string
}
