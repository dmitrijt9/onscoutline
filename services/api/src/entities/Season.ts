import { Column, Entity, PrimaryColumn } from 'typeorm'

@Entity()
export class Season {
    /**
     * Name of the season should be unique by its principle
     */
    @PrimaryColumn()
    name: string

    @Column()
    year: string

    @Column('varchar')
    type: SeasonType
}

export enum SeasonType {
    Spring = 'Spring',
    Autumn = 'Autumn',
}
