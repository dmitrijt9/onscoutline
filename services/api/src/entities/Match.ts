import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm'
import { Club } from './Club'
import { CompetitionHasSeason } from './Relations/CompetitionHasSeason'
import { ISO8601 } from './types'

@Entity()
export class Match {
    @PrimaryGeneratedColumn()
    id: number

    @Column('datetime')
    when: ISO8601

    @Column({ nullable: true })
    where?: string

    @ManyToOne(() => Club)
    homeTeam: Club

    @ManyToOne(() => Club)
    awayTeam: Club

    @Column('int')
    scoreHome: number

    @Column('int')
    scoreAway: number

    @ManyToOne(() => CompetitionHasSeason, { nullable: true })
    competitionSeason?: CompetitionHasSeason
}
