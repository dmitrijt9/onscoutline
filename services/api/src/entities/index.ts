import { Club } from './Club'
import { Competition } from './Competition'
import { Match } from './Match'
import { Player } from './Player'
import { CompetitionHasSeason } from './Relations/CompetitionHasSeason'
import { PlayerInClub } from './Relations/PlayerInClub'
import { PlayerInMatch } from './Relations/PlayerInMatch'
import { Season } from './Season'

export const entities = [
    Club,
    Competition,
    Player,
    PlayerInClub,
    Match,
    PlayerInMatch,
    Season,
    CompetitionHasSeason,
]
