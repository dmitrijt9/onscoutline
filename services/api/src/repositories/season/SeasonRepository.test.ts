import { Container } from '../../dependency/container/index'
import {
    cleanDb,
    createTestingClient,
    stopTestApplication,
    TestingClient,
} from '../../dependency/test-utils/index'
import { SeasonType } from '../../entities/Season'

describe('Season Repository', () => {
    let testingClient: TestingClient
    let container: Container

    beforeAll(async () => {
        testingClient = await createTestingClient()
        container = testingClient.container
    })

    afterAll(async () => {
        await stopTestApplication(testingClient.application)
    })

    beforeEach(async () => {
        await cleanDb(testingClient)
    })

    it('should save a multiple seasons without problems', async () => {
        const { seasonRepository } = container

        const savedSeasons = await seasonRepository.save([
            {
                name: 'jaro2020',
                type: SeasonType.Spring,
                year: '2020',
            },
            {
                name: 'podzim2020',
                type: SeasonType.Autumn,
                year: '2020',
            },
            {
                name: 'jaro2021',
                type: SeasonType.Spring,
                year: '2021',
            },
            {
                name: 'podzim2021',
                type: SeasonType.Autumn,
                year: '2021',
            },
        ])

        expect(savedSeasons.length).toBe(4)
        expect(savedSeasons).toMatchSnapshot()
    })
})
