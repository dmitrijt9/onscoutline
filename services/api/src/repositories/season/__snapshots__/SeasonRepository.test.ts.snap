// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Season Repository should save a multiple seasons without problems 1`] = `
Array [
  Object {
    "name": "jaro2020",
    "type": "Spring",
    "year": "2020",
  },
  Object {
    "name": "podzim2020",
    "type": "Autumn",
    "year": "2020",
  },
  Object {
    "name": "jaro2021",
    "type": "Spring",
    "year": "2021",
  },
  Object {
    "name": "podzim2021",
    "type": "Autumn",
    "year": "2021",
  },
]
`;
